src/bidiMapper/domains/context/BrowsingContextProcessor.ts

```diff
@@ -335,89 +336,136 @@ export class BrowsingContextProcessor {
+        );
 
-    const cdpTarget = CdpTarget.create(
-      targetInfo.targetId,
-      targetCdpClient,
-      this.#browserCdpClient,
-      sessionId,
-      this.#realmStorage,
-      this.#eventManager,
-      this.#preloadScriptStorage,
-      this.#networkStorage,
-      this.#acceptInsecureCerts
-    );
+        this.#createWorkerRealm(cdpTarget);
+        return;
+      }
+    }
 
-    if (maybeContext) {
-      // OOPiF.
-      maybeContext.updateCdpTarget(cdpTarget);
-    } else {
-      // New context.
-      BrowsingContextImpl.create(
-        cdpTarget,
+    // DevTools or some other not supported by BiDi target. Just release
+    // debugger and ignore them.
+    targetCdpClient
+      .sendCommand('Runtime.runIfWaitingForDebugger')
+      .then(() =>
+        parentSessionCdpClient.sendCommand('Target.detachFromTarget', params)
+      )
+      .catch((error) => this.#logger?.(LogType.debugError, error));
+  }
+
+  #workers = new Map<string, Realm>();
+  #createWorkerRealm(cdpTarget: CdpTarget) {
+    cdpTarget.cdpClient.on('Runtime.executionContextCreated', (params) => {
+      const {uniqueId, id, origin} = params.context;
+      const realm = new Realm(
         this.#realmStorage,
-        targetInfo.targetId,
-        null,
-        this.#eventManager,
         this.#browsingContextStorage,
+        uniqueId,
+        cdpTarget.targetId,
+        id,
+        serializeOrigin(origin),
+        'dedicated-worker',
+        undefined,
+        cdpTarget.cdpClient,
+        this.#eventManager,
         this.#logger
       );
-    }
+      this.#workers.set(cdpTarget.cdpSessionId, realm);
+    });
   }
 
   #handleDetachedFromTargetEvent(
     params: Protocol.Target.DetachedFromTargetEvent
   ) {
-    // XXX: params.targetId is deprecated. Update this class to track using
-    // params.sessionId instead.
-    // https://github.com/GoogleChromeLabs/chromium-bidi/issues/60
-    const contextId = params.targetId!;
-    this.#browsingContextStorage.findContext(contextId)?.dispose();
-
-    this.#preloadScriptStorage
-      .find({targetId: contextId})
-      .map((preloadScript) => preloadScript.dispose(contextId));
+    const context = this.#browsingContextStorage
+      .getAllContexts()
+      .find((context) => context.cdpTarget.cdpSessionId === params.sessionId);
+    if (context) {
+      context.dispose();
+      this.#preloadScriptStorage
+        .find({targetId: context.id})
+        .map((preloadScript) => preloadScript.dispose(context.id));
+      return;
+    }
+
+    const worker = this.#workers.get(params.sessionId);
+    if (worker) {
+      this.#realmStorage.deleteRealms({
+        cdpSessionId: worker.cdpClient.sessionId,
+      });
+    }
   }
 
   #handleTargetInfoChangedEvent(
     params: Protocol.Target.TargetInfoChangedEvent
   ) {
-    const contextId = params.targetInfo.targetId;
-    this.#browsingContextStorage
-      .findContext(contextId)
-      ?.onTargetInfoChanged(params);
-  }
```

Comment:
nit: any particular reason for changing the method?

Reply:
Clarity